pkg/crd/gen.go

```diff
@@ -181,6 +185,49 @@ func (g Generator) Generate(ctx *genall.GenerationContext) error {
 	return nil
 }
 
+func removeDescriptionFromMetadata(crd *apiext.CustomResourceDefinition) {
+	for _, versionSpec := range crd.Spec.Versions {
+		if versionSpec.Schema != nil {
+			removeDescriptionFromMetadataProps(versionSpec.Schema.OpenAPIV3Schema)
+		}
+	}
+}
+
+func removeDescriptionFromMetadataProps(v *apiext.JSONSchemaProps) {
+	if m, ok := v.Properties["metadata"]; ok {
+		meta := &m
+		if meta.Description != "" {
+			fmt.Fprintf(os.Stderr, "Warning: metadata description unsupported. Removing description: %s\n", meta.Description)
```

Comment:
Hi, I test your code in myself environment but it don't work well as lack of consider the other fields which may container the struct `JSONSchemaProps`also. I try to changed the code as follow and it works. That's may be a suggestion.
```
	if v == nil {
		return
	}
	for str, property := range v.Properties {
		if str == "metadata" {
			if property.Description != "" {
				fmt.Fprintf(os.Stderr, "Warning: version metadata description unsupported. Removing description: %s\n", property.Description)
				property.Description = ""
			}
		}
		if property.Items != nil {
			removeDescriptionFromMetadataProps(property.Items.Schema)
			for i := range property.Items.JSONSchemas {
				removeDescriptionFromMetadataProps(&property.Items.JSONSchemas[i])
			}
		}
		if property.AdditionalProperties != nil {
			removeDescriptionFromMetadataProps(property.AdditionalProperties.Schema)
		}
		if property.AdditionalItems != nil {
			removeDescriptionFromMetadataProps(property.AdditionalItems.Schema)
		}
		removeDescriptionFromMetadataProps(&property)
		v.Properties[str] = property
	}
```

Reply:
friendly ping @kevindelgado

Reply:
Thanks!

Reply:
top-level metadata shouldn't have any fields -- it's explicitly stripped because the kubernetes API server replaces/ignores most of it.  

AFAIK, the API server only enforces this constraint on the top-level metadata field, not any embedded metadata, so it should be sufficient to simply clear description from the the root metadata field, and leave it at that.

@whalecold if this doesn't work, can you please provide an example that's giving you issues

Reply:
I think this warning might actually just confuse people too, so prob best to remove it -- it's gonna hit a lot of codebases and doesn't really have a material impact on them.